src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -35,6 +35,8 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+import javax.annotation.Nonnull;
+
```

Comment:
This is not the annotation checker framework works with. I'd suggest reading the checker documentation. It is a complex topic, but achievable by breaking the problem into bits. https://checkerframework.org/manual/#nullness-checker

Reply:
> This is not the annotation checker framework works with

@NonNull is a  default or expected value. So I think using @NonNull annotation makes no sense. Letme study more on this topic to understand it better.

Reply:
Two annotations with the same name don't mean they are the same. Checker framework expects this annotation: https://checkerframework.org/api/org/checkerframework/checker/nullness/qual/NonNull.html. You are importing the wrong one.

Reply:
This annotation will work with Checker, no need to use the Checker one. In short, this is fine

Reply:
https://checkerframework.org/manual/#nullness-related-work

>If your code is already annotated with a different nullness annotation, the Checker Framework can type-check your code. It treats annotations from other tools as if you had written the corresponding annotation from the Nullness Checker

> javax.annotation.Nonnull ‍ ⇒  ‍org.checkerframework.checker.nullness.qual.NonNull ‍

why to use checker annotation ?

Reply:
https://checkerframework.org/api/org/checkerframework/checker/nullness/qual/NonNull.html
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/annotation/RetentionPolicy.html#RUNTIME

this means we have to keep Checkeras our compile dependency that grow size of `-all` jar, as users do not have benefits.

Reply:
Yeah, that's why I added:

> This annotation will work with Checker, no need to use the Checker one. In short, this is fine

Reply:
> https://checkerframework.org/api/org/checkerframework/checker/nullness/qual/NonNull.html https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/annotation/RetentionPolicy.html#RUNTIME
> 
> this means we have to keep Checkeras our compile dependency that grow size of `-all` jar, as users do not have benefits.

Cool means using javax.annotation.Nonnull will be okay.

Reply:
Yes